backend/main.py

```
@@ -118,3 +118,49 @@ def read_sales_by_year(year: int, session: SessionDep):
+    except Exception as e:
+        session.rollback()
+        raise HTTPException(status_code=500, detail=f"Failed to create word: {str(e)}")
+
+
+@app.get("/words/", response_model=list[schemas.Word])
+def read_words(session: SessionDep, skip: int = 0, limit: int = 10):
+    try:
+        words = crud.get_words(session, skip, limit)
+        return words
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Failed to read words: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)